src/app/client/src/app/modules/dashboard/components/course-dashboard/course-dashboard.component.ts

```diff
@@ -1,28 +1,102 @@
+import { ActivatedRoute, Router } from '@angular/router';
+import {  } from 'rxjs';
+import { takeUntil } from 'rxjs/operators';
+
+enum dashBoardTitles {
+  batches = 'TOTAL BATCHES',
+  enrollments = 'TOTAL ENROLLMENTS',
+  completions = 'TOTAL COMPLETIONS',
+}
 @Component({
   selector: 'app-course-dashboard',
   templateUrl: './course-dashboard.component.html',
   styleUrls: ['./course-dashboard.component.scss']
 })
 export class CourseDashboardComponent implements OnInit, OnDestroy {
-  batchList = [];
-  dashBoardData: IDashboard;
+
+  public dashBoardSmallCards: IDashboard = {
+    totalBatches: {
+      title: dashBoardTitles.batches,
+      count: 0
+    },
+    totalEnrollment: {
+      title: dashBoardTitles.enrollments,
+      count: 0
+    }
+  };
+
+  public dashBoardLargeCards: IDashBoardLarge = {
+    totalCompleted: {
+      title: dashBoardTitles.completions,
+      count: 0
+    }
+  };
+
   public unsubscribe$ = new Subject<void>();
+  courseId: string;
+  queryParams;
+  telemetryImpression: IImpressionEventInput;
 
-  constructor(private courseProgressService: CourseProgressService) { }
+  constructor(
+    private courseProgressService: CourseProgressService,
+    private activatedRoute: ActivatedRoute,
+    private userService: UserService,
+    private router: Router,
+    private navigationhelperService: NavigationHelperService
+    ) { }
 
   ngOnInit() {
-  this.getBatchList();
+    this.courseId = _.get(this.activatedRoute.snapshot.params, 'courseId');
+    this.queryParams = this.activatedRoute.snapshot.queryParams;
+    this.getBatchList();
+    this.setImpressionEvent();
+  }
+
+  setImpressionEvent() {
+    this.telemetryImpression = {
+      context: {
+        env: _.get(this.activatedRoute.snapshot, 'data.telemetry.env')
+      },
+      edata: {
+        type: _.get(this.activatedRoute.snapshot, 'data.telemetry.type'),
+        pageid: _.get(this.activatedRoute.snapshot, 'data.telemetry.pageid'),
+        uri: this.router.url,
+        duration: this.navigationhelperService.getPageLoadTime()
+      }
+    };
   }
 
   getBatchList() {
+    const searchParams = {
+      courseId: this.courseId,
+      status: ['0', '1', '2'],
+      createdBy: this.userService.userid
+    };
+    this.courseProgressService.getBatches(searchParams).pipe(takeUntil(this.unsubscribe$)).subscribe(data => {
+      this.getEnrollmentAndCompletedCount(_.get(data, 'result.response'));
+    }, err => {
+    });
+
   }
 
-  getEnrollmentAndCompletedCount() {
+  getEnrollmentAndCompletedCount(batchList) {
+    const batches = _.get(batchList, 'content');
+    if (batches) {
+      this.dashBoardSmallCards.totalBatches.count = _.get(batchList, 'count');
+      _.forEach(batches, batch => {
+        this.dashBoardLargeCards.totalCompleted.count += _.get(batch, 'completedCount');
+        this.dashBoardSmallCards.totalEnrollment.count += _.get(batch, 'participantCount');
+      });
+    }
```

Comment:
test cases for this method as well. and add DOC comments for each and every method. will be easier to understand.